src/spend.rs

```diff
@@ -295,13 +302,20 @@ fn select_coins_for_spend(
         output_weight: {
             let mut tx_with_change = base_tx;
             tx_with_change.output.push(change_txo);
-            tx_with_change
+            let weight: u32 = tx_with_change
                 .weight()
                 .to_wu()
                 .checked_sub(base_weight.into())
                 .expect("base_weight can't be larger")
                 .try_into()
-                .expect("tx size must always fit in u32")
+                .expect("tx size must always fit in u32");
+            // Starting with version 0.31, rust-bitcoin now accounts for the segwit marker when
+            // serializing transactions with no input. See comment above.
+            if tx_with_change.input.is_empty() {
+                weight.saturating_sub(2)
+            } else {
+                weight
+            }
```

Comment:
I think we could alternatively subtract `base_tx.weight().to_wu()` instead of `base_weight` and then the extra 2 would be cancelled out from both weight values:
```diff
diff --git a/src/spend.rs b/src/spend.rs
index a373542..fc71280 100644
--- a/src/spend.rs
+++ b/src/spend.rs
@@ -300,22 +300,15 @@ fn select_coins_for_spend(
     let long_term_feerate = FeeRate::from_sat_per_vb(LONG_TERM_FEERATE_VB);
     let drain_weights = DrainWeights {
         output_weight: {
-            let mut tx_with_change = base_tx;
+            let mut tx_with_change = base_tx.clone();
             tx_with_change.output.push(change_txo);
-            let weight: u32 = tx_with_change
+            tx_with_change
                 .weight()
                 .to_wu()
-                .checked_sub(base_weight.into())
+                .checked_sub(base_tx.weight().to_wu())
                 .expect("base_weight can't be larger")
                 .try_into()
-                .expect("tx size must always fit in u32");
-            // Starting with version 0.31, rust-bitcoin now accounts for the segwit marker when
-            // serializing transactions with no input. See comment above.
-            if tx_with_change.input.is_empty() {
-                weight.saturating_sub(2)
-            } else {
-                weight
-            }
+                .expect("tx size must always fit in u32")
         },
         spend_weight: max_input_weight,
     };

```

Reply:
Good call. Did something similar:
```diff
diff --git a/src/spend.rs b/src/spend.rs
index e8e2e10..091839f 100644
--- a/src/spend.rs
+++ b/src/spend.rs
@@ -300,22 +300,18 @@ fn select_coins_for_spend(
     let long_term_feerate = FeeRate::from_sat_per_vb(LONG_TERM_FEERATE_VB);
     let drain_weights = DrainWeights {
         output_weight: {
+            // We don't reuse the above base_weight.since 2 WU may have been substracted from it.
+            // See comment above for details.
+            let nochange_weight = base_tx.weight().to_wu();
             let mut tx_with_change = base_tx;
             tx_with_change.output.push(change_txo);
-            let weight: u32 = tx_with_change
+            tx_with_change
                 .weight()
                 .to_wu()
-                .checked_sub(base_weight.into())
+                .checked_sub(nochange_weight)
                 .expect("base_weight can't be larger")
                 .try_into()
-                .expect("tx size must always fit in u32");
-            // Starting with version 0.31, rust-bitcoin now accounts for the segwit marker when
-            // serializing transactions with no input. See comment above.
-            if tx_with_change.input.is_empty() {
-                weight.saturating_sub(2)
-            } else {
-                weight
-            }
+                .expect("tx size must always fit in u32")
         },
         spend_weight: max_input_weight,
     };
```

(Using a different var name to avoid shadowing if ever refactoring this code in the future.)